core/domain/classifier_domain.py

```diff
@@ -170,3 +169,191 @@ def validate(self):
+                    'answers': ['a2', 'a3']
+                }
+            ]
+        """
+        self._job_id = job_id
+        self._algorithm_id = algorithm_id
+        self._exp_id = exp_id
+        self._exp_version_when_created = exp_version_when_created
+        self._state_name = state_name
+        self._status = status
+        self._training_data = copy.deepcopy(training_data)
+
+    @property
+    def job_id(self):
+        return self._job_id
+
+    @property
+    def algorithm_id(self):
+        return self._algorithm_id
+
+    @property
+    def exp_id(self):
+        return self._exp_id
+
+    @property
+    def exp_version_when_created(self):
+        return self._exp_version_when_created
+
+    @property
+    def state_name(self):
+        return self._state_name
+
+    @property
+    def status(self):
+        return self._status
+
+    @property
+    def training_data(self):
+        return self._training_data
+
+    def to_dict(self):
+        """Constructs a dict representation of training job domain object.
+
+        Returns:
+            A dict representation of training job domain object.
+        """
+
+        return {
+            'job_id': self._job_id,
+            'algorithm_id': self._algorithm_id,
+            'exp_id': self._exp_id,
+            'exp_version_when_created': self._exp_version_when_created,
+            'state_name': self._state_name,
+            'status': self._status,
+            'training_data': self._training_data
+        }
+
+    def validate(self):
+        """Validates the training job before it is saved to storage."""
+
+        algorithm_ids = []
+        if not isinstance(self.job_id, basestring):
+            raise utils.ValidationError(
+                'Expected id to be a string, received %s' % self.job_id)
+
+        if not isinstance(self.exp_id, basestring):
+            raise utils.ValidationError(
+                'Expected exp_id to be a string, received %s' % self.exp_id)
+
+        if not isinstance(self.exp_version_when_created, int):
+            raise utils.ValidationError(
+                'Expected exp_version_when_created to be an int, received %s' %
+                self.exp_version_when_created)
+
+        if not isinstance(self.state_name, basestring):
+            raise utils.ValidationError(
+                'Expected state to be a string, received %s' % self.state_name)
+        utils.require_valid_name(self.state_name, 'the state name')
+
+        if self.status not in feconf.ALLOWED_TRAINING_JOB_STATUSES:
+            raise utils.ValidationError(
+                'Expected status to be in %s, received %s' %
+                feconf.ALLOWED_TRAINING_JOB_STATUSES,
+                self.exp_version_when_created)
+
+        if not isinstance(self.algorithm_id, basestring):
+            raise utils.ValidationError(
+                'Expected algorithm_id to be a string, received %s' %
+                self.algorithm_id)
+        utils.require_valid_name(
```

Comment:
FYI this seems somewhat optional; it just needs to be in the list of algorithm ids.